tests/test_builtin_casters.cpp

```diff
@@ -113,6 +114,8 @@ TEST_SUBMODULE(builtin_casters, m) {
     // test_single_char_arguments
     m.def("ord_char8", [](char8_t c) -> int { return static_cast<unsigned char>(c); });
     m.def("ord_char8_lv", [](char8_t &c) -> int { return static_cast<unsigned char>(c); });
+#else  // PYBIND11_HAS_U8STRING
```

Comment:
I really like this PR, but find it unfortunate that the true/false assignments are separated in space. How about:
```
#ifndef PYBIND11_HAS_U8STRING
    m.attr("has_u8string") = false;
#else
    m.attr("has_u8string") = true;
    ...  
#endif
```

Reply:
Starting with negative and then putting the bulk in a double negative is not really ideal, either? How about:

```cpp
#ifdef PYBIND11_HAS_U8STRING
    ...
    m.attr("has_u8string") = true;
#else
    m.attr("has_u8string") = false;
#endif
```

Reply:
Looks good. Either way is fine with me.